### DISCUSSION TITLE
Making basic example work

### DISCUSSION THREAD

**Discussion by nekopep** [EtiennePerot/open-webui-code-execution#31] [Q&A] [answered]
Hello,
I've started to try to work with the plugin but i'm facing a issue, models htat I tried (Mistral Large and ChatGPT 4o) always want to run python code and fail because they are missing module.
I tried a sentence like this:
plot the stock price of Apple, Tesla, Nvidia and Microsoft YTD 
![image](https://github.com/user-attachments/assets/785fcd44-bac1-46b6-914d-12dcc122e506)

I tried: 
install all module needed with bash and execude python code to plot the stock price of Apple, Tesla, Nvidia and Microsoft YTD
Bu tmodel does not seem to be able to chain the tool calling:
![image](https://github.com/user-attachments/assets/c029c7eb-b4e4-4067-964a-2982d92022ed)

I even try to ask first install an dthen excute without success:
![image](https://github.com/user-attachments/assets/b93ca7fb-ac8d-41c7-a9a6-85f9fbe23624)

Do I miss something, did I misconfigured something mandatory?

**Comment by EtiennePerot** [comment, marked as answer]
You are not missing anything. The problem is that each code execution sandbox is, well, a sandbox. That means they are isolated from each other. Packages installed in one sandbox will not be present in the others. Enabling sandboxes to live across code executions is being worked on in issue #15.
